cmake/3.9.0/share/man/man1/cmake.1
.\" Man page generated from reStructuredText.
.
.TH "CMAKE" "1" "Jul 29, 2017" "3.9.0" "CMake"
.SH NAME
cmake \- CMake Command-Line Reference
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
cmake [<options>] (<path\-to\-source> | <path\-to\-existing\-build>)
cmake [(\-D <var>=<value>)...] \-P <cmake\-script\-file>
cmake \-\-build <dir> [<options>...] [\-\- <build\-tool\-options>...]
cmake \-E <command> [<options>...]
cmake \-\-find\-package <options>...
.ft P
.fi
.UNINDENT
.UNINDENT
.SH DESCRIPTION
.sp
The “cmake” executable is the CMake command\-line interface.  It may be
used to configure projects in scripts.  Project configuration settings
may be specified on the command line with the \-D option.
.sp
CMake is a cross\-platform build system generator.  Projects specify
their build process with platform\-independent CMake listfiles included
in each directory of a source tree with the name CMakeLists.txt.
Users build a project by using CMake to generate a build system for a
native tool on their platform.
.SH OPTIONS
.INDENT 0.0
.TP
.B \fB\-C <initial\-cache>\fP
Pre\-load a script to populate the cache.
.sp
When cmake is first run in an empty build tree, it creates a
CMakeCache.txt file and populates it with customizable settings for
the project.  This option may be used to specify a file from which
to load cache entries before the first pass through the project’s
cmake listfiles.  The loaded entries take priority over the
project’s default values.  The given file should be a CMake script
containing SET commands that use the CACHE option, not a
cache\-format file.
.TP
.B \fB\-D <var>:<type>=<value>, \-D <var>=<value>\fP
Create a cmake cache entry.
.sp
When cmake is first run in an empty build tree, it creates a
CMakeCache.txt file and populates it with customizable settings for
the project.  This option may be used to specify a setting that
takes priority over the project’s default value.  The option may be
repeated for as many cache entries as desired.
.sp
If the \fB:<type>\fP portion is given it must be one of the types
specified by the \fBset()\fP command documentation for its
\fBCACHE\fP signature.
If the \fB:<type>\fP portion is omitted the entry will be created
with no type if it does not exist with a type already.  If a
command in the project sets the type to \fBPATH\fP or \fBFILEPATH\fP
then the \fB<value>\fP will be converted to an absolute path.
.sp
This option may also be given as a single argument:
\fB\-D<var>:<type>=<value>\fP or \fB\-D<var>=<value>\fP\&.
.TP
.B \fB\-U <globbing_expr>\fP
Remove matching entries from CMake cache.
.sp
This option may be used to remove one or more variables from the
CMakeCache.txt file, globbing expressions using * and ? are
supported.  The option may be repeated for as many cache entries as
desired.
.sp
Use with care, you can make your CMakeCache.txt non\-working.
.TP
.B \fB\-G <generator\-name>\fP
Specify a build system generator.
.sp
CMake may support multiple native build systems on certain
platforms.  A generator is responsible for generating a particular
build system.  Possible generator names are specified in the
\fBcmake\-generators(7)\fP manual.
.TP
.B \fB\-T <toolset\-spec>\fP
Toolset specification for the generator, if supported.
.sp
Some CMake generators support a toolset specification to tell
the native build system how to choose a compiler.  See the
\fBCMAKE_GENERATOR_TOOLSET\fP variable for details.
.TP
.B \fB\-A <platform\-name>\fP
Specify platform name if supported by generator.
.sp
Some CMake generators support a platform name to be given to the
native build system to choose a compiler or SDK.  See the
\fBCMAKE_GENERATOR_PLATFORM\fP variable for details.
.TP
.B \fB\-Wno\-dev\fP
Suppress developer warnings.
.sp
Suppress warnings that are meant for the author of the
CMakeLists.txt files. By default this will also turn off
deprecation warnings.
.TP
.B \fB\-Wdev\fP
Enable developer warnings.
.sp
Enable warnings that are meant for the author of the CMakeLists.txt
files. By default this will also turn on deprecation warnings.
.TP
.B \fB\-Werror=dev\fP
Make developer warnings errors.
.sp
Make warnings that are meant for the author of the CMakeLists.txt files
errors. By default this will also turn on deprecated warnings as errors.
.TP
.B \fB\-Wno\-error=dev\fP
Make developer warnings not errors.
.sp
Make warnings that are meant for the author of the CMakeLists.txt files not
errors. By default this will also turn off deprecated warnings as errors.
.TP
.B \fB\-Wdeprecated\fP
Enable deprecated functionality warnings.
.sp
Enable warnings for usage of deprecated functionality, that are meant
for the author of the CMakeLists.txt files.
.TP
.B \fB\-Wno\-deprecated\fP
Suppress deprecated functionality warnings.
.sp
Suppress warnings for usage of deprecated functionality, that are meant
for the author of the CMakeLists.txt files.
.TP
.B \fB\-Werror=deprecated\fP
Make deprecated macro and function warnings errors.
.sp
Make warnings for usage of deprecated macros and functions, that are meant
for the author of the CMakeLists.txt files, errors.
.TP
.B \fB\-Wno\-error=deprecated\fP
Make deprecated macro and function warnings not errors.
.sp
Make warnings for usage of deprecated macros and functions, that are meant
for the author of the CMakeLists.txt files, not errors.
.TP
.B \fB\-E <command> [<options>...]\fP
See \fI\%Command\-Line Tool Mode\fP\&.
.TP
.B \fB\-L[A][H]\fP
List non\-advanced cached variables.
.sp
List cache variables will run CMake and list all the variables from
the CMake cache that are not marked as INTERNAL or ADVANCED.  This
will effectively display current CMake settings, which can then be
changed with \-D option.  Changing some of the variables may result
in more variables being created.  If A is specified, then it will
display also advanced variables.  If H is specified, it will also
display help for each variable.
.TP
.B \fB\-\-build <dir>\fP
See \fI\%Build Tool Mode\fP\&.
.TP
.B \fB\-N\fP
View mode only.
.sp
Only load the cache.  Do not actually run configure and generate
steps.
.TP
.B \fB\-P <file>\fP
Process script mode.
.sp
Process the given cmake file as a script written in the CMake
language.  No configure or generate step is performed and the cache
is not modified.  If variables are defined using \-D, this must be
done before the \-P argument.
.TP
.B \fB\-\-find\-package\fP
See \fI\%Find\-Package Tool Mode\fP\&.
.TP
.B \fB\-\-graphviz=[file]\fP
Generate graphviz of dependencies, see CMakeGraphVizOptions.cmake for more.
.sp
Generate a graphviz input file that will contain all the library and
executable dependencies in the project.  See the documentation for
CMakeGraphVizOptions.cmake for more details.
.TP
.B \fB\-\-system\-information [file]\fP
Dump information about this system.
.sp
Dump a wide range of information about the current system.  If run
from the top of a binary tree for a CMake project it will dump
additional information such as the cache, log files etc.
.TP
.B \fB\-\-debug\-trycompile\fP
Do not delete the try_compile build tree. Only useful on one try_compile at a time.
.sp
Do not delete the files and directories created for try_compile
calls.  This is useful in debugging failed try_compiles.  It may
however change the results of the try\-compiles as old junk from a
previous try\-compile may cause a different test to either pass or
fail incorrectly.  This option is best used for one try\-compile at a
time, and only when debugging.
.TP
.B \fB\-\-debug\-output\fP
Put cmake in a debug mode.
.sp
Print extra information during the cmake run like stack traces with
message(send_error ) calls.
.TP
.B \fB\-\-trace\fP
Put cmake in trace mode.
.sp
Print a trace of all calls made and from where.
.TP
.B \fB\-\-trace\-expand\fP
Put cmake in trace mode.
.sp
Like \fB\-\-trace\fP, but with variables expanded.
.TP
.B \fB\-\-trace\-source=<file>\fP
Put cmake in trace mode, but output only lines of a specified file.
.sp
Multiple options are allowed.
.TP
.B \fB\-\-warn\-uninitialized\fP
Warn about uninitialized values.
.sp
Print a warning when an uninitialized variable is used.
.TP
.B \fB\-\-warn\-unused\-vars\fP
Warn about unused variables.
.sp
Find variables that are declared or set, but not used.
.TP
.B \fB\-\-no\-warn\-unused\-cli\fP
Don’t warn about command line options.
.sp
Don’t find variables that are declared on the command line, but not
used.
.TP
.B \fB\-\-check\-system\-vars\fP
Find problems with variable usage in system files.
.sp
Normally, unused and uninitialized variables are searched for only
in CMAKE_SOURCE_DIR and CMAKE_BINARY_DIR.  This flag tells CMake to
warn about other files as well.
.UNINDENT
.INDENT 0.0
.TP
.B \fB\-\-help,\-help,\-usage,\-h,\-H,/?\fP
Print usage information and exit.
.sp
Usage describes the basic command line interface and its options.
.TP
.B \fB\-\-version,\-version,/V [<f>]\fP
Show program name/version banner and exit.
.sp
If a file is specified, the version is written into it.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-full [<f>]\fP
Print all help manuals and exit.
.sp
All manuals are printed in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-manual <man> [<f>]\fP
Print one help manual and exit.
.sp
The specified manual is printed in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-manual\-list [<f>]\fP
List help manuals available and exit.
.sp
The list contains all manuals for which help may be obtained by
using the \fB\-\-help\-manual\fP option followed by a manual name.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-command <cmd> [<f>]\fP
Print help for one command and exit.
.sp
The \fBcmake\-commands(7)\fP manual entry for \fB<cmd>\fP is
printed in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-command\-list [<f>]\fP
List commands with help available and exit.
.sp
The list contains all commands for which help may be obtained by
using the \fB\-\-help\-command\fP option followed by a command name.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-commands [<f>]\fP
Print cmake\-commands manual and exit.
.sp
The \fBcmake\-commands(7)\fP manual is printed in a
human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-module <mod> [<f>]\fP
Print help for one module and exit.
.sp
The \fBcmake\-modules(7)\fP manual entry for \fB<mod>\fP is printed
in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-module\-list [<f>]\fP
List modules with help available and exit.
.sp
The list contains all modules for which help may be obtained by
using the \fB\-\-help\-module\fP option followed by a module name.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-modules [<f>]\fP
Print cmake\-modules manual and exit.
.sp
The \fBcmake\-modules(7)\fP manual is printed in a human\-readable
text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-policy <cmp> [<f>]\fP
Print help for one policy and exit.
.sp
The \fBcmake\-policies(7)\fP manual entry for \fB<cmp>\fP is
printed in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-policy\-list [<f>]\fP
List policies with help available and exit.
.sp
The list contains all policies for which help may be obtained by
using the \fB\-\-help\-policy\fP option followed by a policy name.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-policies [<f>]\fP
Print cmake\-policies manual and exit.
.sp
The \fBcmake\-policies(7)\fP manual is printed in a
human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-property <prop> [<f>]\fP
Print help for one property and exit.
.sp
The \fBcmake\-properties(7)\fP manual entries for \fB<prop>\fP are
printed in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-property\-list [<f>]\fP
List properties with help available and exit.
.sp
The list contains all properties for which help may be obtained by
using the \fB\-\-help\-property\fP option followed by a property name.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-properties [<f>]\fP
Print cmake\-properties manual and exit.
.sp
The \fBcmake\-properties(7)\fP manual is printed in a
human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-variable <var> [<f>]\fP
Print help for one variable and exit.
.sp
The \fBcmake\-variables(7)\fP manual entry for \fB<var>\fP is
printed in a human\-readable text format.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-variable\-list [<f>]\fP
List variables with help available and exit.
.sp
The list contains all variables for which help may be obtained by
using the \fB\-\-help\-variable\fP option followed by a variable name.
The help is printed to a named <f>ile if given.
.TP
.B \fB\-\-help\-variables [<f>]\fP
Print cmake\-variables manual and exit.
.sp
The \fBcmake\-variables(7)\fP manual is printed in a
human\-readable text format.
The help is printed to a named <f>ile if given.
.UNINDENT
.SH BUILD TOOL MODE
.sp
CMake provides a command\-line signature to build an already\-generated
project binary tree:
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
cmake \-\-build <dir> [<options>...] [\-\- <build\-tool\-options>...]
.ft P
.fi
.UNINDENT
.UNINDENT
.sp
This abstracts a native build tool’s command\-line interface with the
following options:
.INDENT 0.0
.TP
.B \fB\-\-build <dir>\fP
Project binary directory to be built.  This is required and must be first.
.TP
.B \fB\-\-target <tgt>\fP
Build \fB<tgt>\fP instead of default targets.  May only be specified once.
.TP
.B \fB\-\-config <cfg>\fP
For multi\-configuration tools, choose configuration \fB<cfg>\fP\&.
.TP
.B \fB\-\-clean\-first\fP
Build target \fBclean\fP first, then build.
(To clean only, use \fB\-\-target clean\fP\&.)
.TP
.B \fB\-\-use\-stderr\fP
Ignored.  Behavior is default in CMake >= 3.0.
.TP
.B \fB\-\-\fP
Pass remaining options to the native tool.
.UNINDENT
.sp
Run \fBcmake \-\-build\fP with no options for quick help.
.SH COMMAND-LINE TOOL MODE
.sp
CMake provides builtin command\-line tools through the signature:
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
cmake \-E <command> [<options>...]
.ft P
.fi
.UNINDENT
.UNINDENT
.sp
Run \fBcmake \-E\fP or \fBcmake \-E help\fP for a summary of commands.
Available commands are:
.INDENT 0.0
.TP
.B \fBcapabilities\fP
Report cmake capabilities in JSON format. The output is a JSON object
with the following keys:
.INDENT 7.0
.TP
.B \fBversion\fP
A JSON object with version information. Keys are:
.INDENT 7.0
.TP
.B \fBstring\fP
The full version string as displayed by cmake \fB\-\-version\fP\&.
.TP
.B \fBmajor\fP
The major version number in integer form.
.TP
.B \fBminor\fP
The minor version number in integer form.
.TP
.B \fBpatch\fP
The patch level in integer form.
.TP
.B \fBsuffix\fP
The cmake version suffix string.
.TP
.B \fBisDirty\fP
A bool that is set if the cmake build is from a dirty tree.
.UNINDENT
.TP
.B \fBgenerators\fP
A list available generators. Each generator is a JSON object with the
following keys:
.INDENT 7.0
.TP
.B \fBname\fP
A string containing the name of the generator.
.TP
.B \fBtoolsetSupport\fP
\fBtrue\fP if the generator supports toolsets and \fBfalse\fP otherwise.
.TP
.B \fBplatformSupport\fP
\fBtrue\fP if the generator supports platforms and \fBfalse\fP otherwise.
.TP
.B \fBextraGenerators\fP
A list of strings with all the extra generators compatible with
the generator.
.UNINDENT
.TP
.B \fBserverMode\fP
\fBtrue\fP if cmake supports server\-mode and \fBfalse\fP otherwise.
.UNINDENT
.TP
.B \fBchdir <dir> <cmd> [<arg>...]\fP
Change the current working directory and run a command.
.TP
.B \fBcompare_files <file1> <file2>\fP
Check if \fB<file1>\fP is same as \fB<file2>\fP\&. If files are the same,
then returns 0, if not it returns 1.
.TP
.B \fBcopy <file>... <destination>\fP
Copy files to \fB<destination>\fP (either file or directory).
If multiple files are specified, the \fB<destination>\fP must be
directory and it must exist. Wildcards are not supported.
.TP
.B \fBcopy_directory <dir>... <destination>\fP
Copy directories to \fB<destination>\fP directory.
If \fB<destination>\fP directory does not exist it will be created.
.TP
.B \fBcopy_if_different <file>... <destination>\fP
Copy files to \fB<destination>\fP (either file or directory) if
they have changed.
If multiple files are specified, the \fB<destination>\fP must be
directory and it must exist.
.TP
.B \fBecho [<string>...]\fP
Displays arguments as text.
.TP
.B \fBecho_append [<string>...]\fP
Displays arguments as text but no new line.
.TP
.B \fBenv [\-\-unset=NAME]... [NAME=VALUE]... COMMAND [ARG]...\fP
Run command in a modified environment.
.TP
.B \fBenvironment\fP
Display the current environment variables.
.TP
.B \fBmake_directory <dir>...\fP
Create \fB<dir>\fP directories.  If necessary, create parent
directories too.  If a directory already exists it will be
silently ignored.
.TP
.B \fBmd5sum <file>...\fP
Create MD5 checksum of files in \fBmd5sum\fP compatible format:
.INDENT 7.0
.INDENT 3.5
.sp
.nf
.ft C
351abe79cd3800b38cdfb25d45015a15  file1.txt
052f86c15bbde68af55c7f7b340ab639  file2.txt
.ft P
.fi
.UNINDENT
.UNINDENT
.TP
.B \fBremove [\-f] <file>...\fP
Remove the file(s). If any of the listed files already do not
exist, the command returns a non\-zero exit code, but no message
is logged. The \fB\-f\fP option changes the behavior to return a
zero exit code (i.e. success) in such situations instead.
.TP
.B \fBremove_directory <dir>\fP
Remove a directory and its contents.  If a directory does
not exist it will be silently ignored.
.TP
.B \fBrename <oldname> <newname>\fP
Rename a file or directory (on one volume).
.TP
.B \fBserver\fP
Launch \fBcmake\-server(7)\fP mode.
.TP
.B \fBsleep <number>...\fP
Sleep for given number of seconds.
.TP
.B \fBtar [cxt][vf][zjJ] file.tar [<options>...] [\-\-] [<file>...]\fP
Create or extract a tar or zip archive.  Options are:
.INDENT 7.0
.TP
.B \fB\-\-\fP
Stop interpreting options and treat all remaining arguments
as file names even if they start in \fB\-\fP\&.
.TP
.B \fB\-\-files\-from=<file>\fP
Read file names from the given file, one per line.
Blank lines are ignored.  Lines may not start in \fB\-\fP
except for \fB\-\-add\-file=<name>\fP to add files whose
names start in \fB\-\fP\&.
.TP
.B \fB\-\-mtime=<date>\fP
Specify modification time recorded in tarball entries.
.TP
.B \fB\-\-format=<format>\fP
Specify the format of the archive to be created.
Supported formats are: \fB7zip\fP, \fBgnutar\fP, \fBpax\fP,
\fBpaxr\fP (restricted pax, default), and \fBzip\fP\&.
.UNINDENT
.TP
.B \fBtime <command> [<args>...]\fP
Run command and return elapsed time.
.TP
.B \fBtouch <file>\fP
Touch a file.
.TP
.B \fBtouch_nocreate <file>\fP
Touch a file if it exists but do not create it.  If a file does
not exist it will be silently ignored.
.UNINDENT
.SS UNIX\-specific Command\-Line Tools
.sp
The following \fBcmake \-E\fP commands are available only on UNIX:
.INDENT 0.0
.TP
.B \fBcreate_symlink <old> <new>\fP
Create a symbolic link \fB<new>\fP naming \fB<old>\fP\&.
.UNINDENT
.sp
\fBNOTE:\fP
.INDENT 0.0
.INDENT 3.5
Path to where \fB<new>\fP symbolic link will be created has to exist beforehand.
.UNINDENT
.UNINDENT
.SS Windows\-specific Command\-Line Tools
.sp
The following \fBcmake \-E\fP commands are available only on Windows:
.INDENT 0.0
.TP
.B \fBdelete_regv <key>\fP
Delete Windows registry value.
.TP
.B \fBenv_vs8_wince <sdkname>\fP
Displays a batch file which sets the environment for the provided
Windows CE SDK installed in VS2005.
.TP
.B \fBenv_vs9_wince <sdkname>\fP
Displays a batch file which sets the environment for the provided
Windows CE SDK installed in VS2008.
.TP
.B \fBwrite_regv <key> <value>\fP
Write Windows registry value.
.UNINDENT
.SH FIND-PACKAGE TOOL MODE
.sp
CMake provides a helper for Makefile\-based projects with the signature:
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
cmake \-\-find\-package <options>...
.ft P
.fi
.UNINDENT
.UNINDENT
.sp
This runs in a pkg\-config like mode.
.sp
Search a package using \fBfind_package()\fP and print the resulting flags
to stdout.  This can be used to use cmake instead of pkg\-config to find
installed libraries in plain Makefile\-based projects or in autoconf\-based
projects (via \fBshare/aclocal/cmake.m4\fP).
.sp
\fBNOTE:\fP
.INDENT 0.0
.INDENT 3.5
This mode is not well\-supported due to some technical limitations.
It is kept for compatibility but should not be used in new projects.
.UNINDENT
.UNINDENT
.SH SEE ALSO
.sp
The following resources are available to get help using CMake:
.INDENT 0.0
.TP
.B Home Page
\fI\%https://cmake.org\fP
.sp
The primary starting point for learning about CMake.
.TP
.B Frequently Asked Questions
\fI\%https://cmake.org/Wiki/CMake_FAQ\fP
.sp
A Wiki is provided containing answers to frequently asked questions.
.TP
.B Online Documentation
\fI\%https://cmake.org/documentation\fP
.sp
Links to available documentation may be found on this web page.
.TP
.B Mailing List
\fI\%https://cmake.org/mailing\-lists\fP
.sp
For help and discussion about using cmake, a mailing list is
provided at \fI\%cmake@cmake.org\fP\&.  The list is member\-post\-only but one
may sign up on the CMake web page.  Please first read the full
documentation at \fI\%https://cmake.org\fP before posting questions to
the list.
.UNINDENT
.SH COPYRIGHT
2000-2017 Kitware, Inc. and Contributors
.\" Generated by docutils manpage writer.
.


cmake/3.9.0/.brew/cmake.rb
class Cmake < Formula
  desc "Cross-platform make"
  homepage "https://www.cmake.org/"
  url "https://cmake.org/files/v3.9/cmake-3.9.0.tar.gz"
  sha256 "167701525183dbb722b9ffe69fb525aa2b81798cf12f5ce1c020c93394dfae0f"
  head "https://cmake.org/cmake.git"

  option "without-docs", "Don't build man pages"
  option "with-completion", "Install Bash completion (Has potential problems with system bash)"

  depends_on "sphinx-doc" => :build if build.with? "docs"

  # The `with-qt` GUI option was removed due to circular dependencies if
  # CMake is built with Qt support and Qt is built with MySQL support as MySQL uses CMake.
  # For the GUI application please instead use `brew cask install cmake`.

  def install
    args = %W[
      --prefix=#{prefix}
      --no-system-libs
      --parallel=#{ENV.make_jobs}
      --datadir=/share/cmake
      --docdir=/share/doc/cmake
      --mandir=/share/man
      --system-zlib
      --system-bzip2
      --system-curl
    ]

    if build.with? "docs"
      # There is an existing issue around macOS & Python locale setting
      # See https://bugs.python.org/issue18378#msg215215 for explanation
      ENV["LC_ALL"] = "en_US.UTF-8"
      args << "--sphinx-man" << "--sphinx-build=#{Formula["sphinx-doc"].opt_bin}/sphinx-build"
    end

    system "./bootstrap", *args
    system "make"
    system "make", "install"

    if build.with? "completion"
      cd "Auxiliary/bash-completion/" do
        bash_completion.install "ctest", "cmake", "cpack"
      end
    end

    elisp.install "Auxiliary/cmake-mode.el"
  end

  test do
    (testpath/"CMakeLists.txt").write("find_package(Ruby)")
    system bin/"cmake", "."
  end
end
